### PR TITLE
Tests/e2e tests staging

### LINKED ISSUE #1932: Chain regression E2E Tests for btcli
### Summary 

The purpose of these e2e tests is full regression testing of chain features with `btcli`.

### List of commands

**subnets** https://github.com/opentensor/bittensor/pull/1948
- [ ] list - List all subnets on the network
- [ ] metagraph - View a subnet metagraph information, with --netuid 
- [ ] lock_cost - Return the lock cost to register a subnet
- [ ] create - Create a new bittensor subnetwork on this chain.
- [ ] register - Register a wallet to a network, with --netuid
- [ ] hyperparameters - View subnet hyperparameters, with --netuid

**root**
- [x] list - List the root network
- [x] weights - Set weights for root network, with netuid list and weigths list
- [x] get_weights - Get weights for root network.
- [x] boost - Boost weight for a specific subnet by increase amount, with --netuid and --increase
- [ ] slash - Slash weight for a specific subnet by decrease amount, with --netuid and --decrease
- [x] senate_vote - Vote on an active proposal by hash.
- [x] senate - View senate and it's members
- [x] register - Register a wallet to the root network.
- [x] proposals - View active triumvirate proposals and their status
- [x] delegate - Delegate Stake to an account, with --amount
  - [ ] --all (without --amount)
- [x] undelegate          Undelegate Stake from an account, with --amount
  - [ ] --all (without --amount)
- [ ] my_delegates - Show all delegates where I am delegating a positive amount of stake
- [ ] list_delegates - List all delegates on the network
- [ ] list_delegates_lite - List all delegates on the network (lite version).
- [x] nominate - Become a delegate on the network

**wallet**
- [ ] list - List wallets
- [ ] overview - Show registered account overview, with --all
- [ ] transfer - Transfer Tao between accounts.
- [ ] inspect - Inspect a wallet (cold, hot) pair
  - [ ] --all
  - [ ] --netuids
- [ ] balance - Checks the balance of the wallet.
  - [ ] --all
- [ ] set_identity - Create or update identity on-chain for a given cold wallet. Must be a subnet owner.
- [ ] get_identity - Creates a new coldkey (for containing balance) under the specified path.

**stake**
- [x] show - List all stake accounts for wallet.
- [x] add - Add stake to your hotkey accounts from your coldkey.
- [x] remove - Remove stake from the specified hotkey into the coldkey balance.

### Corner cases

The parameters should be tested for positive and negative cases. For example, since netuid is a parameter of `btcli subnet register`, at least two tests should be created:

- test_subnet_register_ok
- test_subnet_register_wrong_netuid_fail

Here are the examples of parameters with types of errors:

- netuid: subnet that doesn't exist
- amount: over balance
- hotkeys: address that is not a hotkey

### PR DESCRIPTION
**Pull request by open-junius**
The PR will fix #1932.
Some explanation for the implementation.
1. currently,  the output of command includes more lines than expected because of bittensor version. We can see the message like `Please update to the latest version at your earliest convenience. Run the following command to upgrade`. that why `assert len(lines) >=` used to check the output.
2. The root network weight set is disable, so some commands are commented in the test case. Will refactor after new implementation is available in bittensor side.
3. The test for delegate stake is commented because we can't get the delegate data via subtensor.get_delegates method.

### CODE REVIEW

File: bittensor/extrinsics/network.py [hunk 1]
```diff
@@ -210,3 +210,109 @@ def set_hyperparameter_extrinsic(
                     f":white_heavy_check_mark: [green]Hyper parameter {parameter} changed to {value}[/green]"
                 )
                 return True
+
```
Comment (distributedstatemachine) [inline bittensor/extrinsics/network.py]: Why have these files changed ? I do not think they do not have anything to do with the pr and should be in the diff

Comment (open-junius) [reply bittensor/extrinsics/network.py]: To register a subnetwork, there is a network_rate_limit. the default value is 7200. I add an extrinsic to set it. then we can register soon after network is up.

File: tests/e2e_tests/subcommands/root/test_root_boost_weights.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
+from bittensor.commands.root import RootSetBoostCommand, RootSetWeightsCommand
+from bittensor.commands.stake import StakeCommand
+from bittensor.commands.unstake import UnStakeCommand
+from bittensor.commands.network import RegisterSubnetworkCommand
+from bittensor.commands.register import RegisterCommand
+from ...utils import new_wallet, sudo_call_set_network_limit
+import bittensor
+
+
+# we can't test it now since the root network weights can't be set
```
Comment (opendansor) [inline tests/e2e_tests/subcommands/root/test_root_boost_weights.py]: If we cant test this right now, should we remove it until we can properly test it?

File: tests/e2e_tests/subcommands/root/test_root_delegate_stake_unstake.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    delegates = local_chain.query(
+        "SubtensorModule",
+        "Delegates",
+        [wallet.hotkey.ss58_address],
+    )
+
+    assert delegates == 11796
+
+    # stake 1 TAO
+    # exec_command(
```
Comment (opendansor) [inline tests/e2e_tests/subcommands/root/test_root_delegate_stake_unstake.py]: What is the purpose of these commented out blocks?

File: tests/e2e_tests/subcommands/root/test_root_get_set_weights.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
+from bittensor.commands.root import RootSetWeightsCommand, RootGetWeightsCommand
+from bittensor.commands.network import RegisterSubnetworkCommand
+from ...utils import new_wallet, sudo_call_set_network_limit
+
+
+# we can't test it now since the root network weights can't be set
```
Comment (opendansor) [inline tests/e2e_tests/subcommands/root/test_root_get_set_weights.py:6]: Does this test run correctly? Maybe we write this test when the command is updated?

File: tests/e2e_tests/subcommands/root/test_root_list.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    (wallet, exec_command) = new_wallet("//Alice", "//Bob")
+
+    exec_command(RootList, ["root", "list"])
+    captured = capsys.readouterr()
+    lines = captured.out.split("\n")
+
+    assert len(lines) >= 4
+    bittensor.logging.info(lines)
+
+    # assert "Root Network" in lines[0]
```
Comment (opendansor) [inline tests/e2e_tests/subcommands/root/test_root_list.py:18]: Why are these lines commented out?

Comment (open-junius) [reply tests/e2e_tests/subcommands/root/test_root_list.py:18]: As I mentioned in the description, we always get the extra output about bittensor version. so just commented out, can add them back after version is aligned.

File: tests/e2e_tests/subcommands/root/test_root_get_set_weights.py (hunk 4 above)
Comment (open-junius) [reply tests/e2e_tests/subcommands/root/test_root_get_set_weights.py:6]: The extrinsic in subtensor side is updated. From @distributedstatemachine , I know cortex team will change accordingly.

File: tests/e2e_tests/subcommands/root/test_root_delegate_stake_unstake.py (hunk 3 above)
Comment (open-junius) [reply tests/e2e_tests/subcommands/root/test_root_delegate_stake_unstake.py]: to add and reduce stake

File: tests/e2e_tests/subcommands/root/test_root_boost_weights.py (hunk 2 above)
Comment (open-junius) [reply tests/e2e_tests/subcommands/root/test_root_boost_weights.py]: pls keep it in the repo. I need some time to try, then know what's the correct parameter and how to check.

Comment (opendansor) [reply tests/e2e_tests/subcommands/root/test_root_boost_weights.py]: Can you keep this in a separate branch? There is no need to push this in while its still under development and not ready to go out to the public. Every test we run on the Bittensor side actually tests what it is supposed to test. This will add tests that don't actually test what they claim to test, and we will be required to "remember" that X or Y test is incomplete, giving us holes in what bugs and issues can go unnoticed.

File: tests/e2e_tests/subcommands/root/test_root_delegate_stake_unstake.py (hunk 3 above)
Comment (opendansor) [reply tests/e2e_tests/subcommands/root/test_root_delegate_stake_unstake.py]: I understand, but the code is commented out... Meaning the test doesn't actually add or reduce stake, and test that stake is added or reduced.

**Review by opendansor — CHANGES_REQUESTED** [review]
Please add tests that actually test what they claim to test. If X or Y test is still under development, keep it in a git branch and work on it as you see fit, but there is no need to push in tests that don't accomplish what they set out to do.

File: tests/e2e_tests/subcommands/root/test_root_delegate_stake_unstake.py (hunk 3 above)
Comment (open-junius) [reply tests/e2e_tests/subcommands/root/test_root_delegate_stake_unstake.py]: Ok, will just keep it in my local branch

File: tests/e2e_tests/subcommands/root/test_root_boost_weights.py (hunk 2 above)
Comment (open-junius) [reply tests/e2e_tests/subcommands/root/test_root_boost_weights.py]: Ok, I will remove it.

**Review by opendansor — APPROVED** [review]

**Review by sam0x17 — APPROVED** [review]
